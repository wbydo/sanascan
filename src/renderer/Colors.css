/* @value primary: #67c5ff; */
@value primary: red;
