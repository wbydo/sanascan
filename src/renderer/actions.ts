import * as types from "./types";

export interface Action {
  type: string;
}

export const increment: () => Action = () => {
  return {
    type: types.INCREMENT,
  };
};
