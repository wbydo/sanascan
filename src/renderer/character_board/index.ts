import reducer from "./reducers";

import * as CharacterBoardType from "./types";

export {CharacterBoardType};

export default reducer;
