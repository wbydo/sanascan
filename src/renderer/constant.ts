export const chars: string[][] = [
  ["あ", "か", "さ", "た", "な", "は", "ま", "や", "ら", "わ"],
  ["い", "き", "し", "ち", "に", "ひ", "み", "",   "り", ""],
  ["う", "く", "す", "つ", "ぬ", "ふ", "む", "ゆ", "る", "を"],
  ["え", "け", "せ", "て", "ね", "へ", "め", "",   "れ", ""],
  ["お", "こ", "そ", "と", "の", "ほ", "も", "よ", "ろ", "ん"],
];

export const MAX_COLUMN_INDEX = chars[0].length - 1;
export const MAX_ROW_INDEX = chars.length - 1;

export const url = "http://localhost:8000";
