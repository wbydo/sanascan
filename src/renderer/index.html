<!DOCTYPE html>
<html lang="ja">
<head>
  <meta charset="utf-8">
  <title>YACOS</title>
  <link rel="stylesheet" href="css/style.css">
</head>
<body>
  <div id="root"></div>
</body>
</html>
