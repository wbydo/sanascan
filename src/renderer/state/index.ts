import { createStore, applyMiddleware } from "redux";
import { combineReducers, Dispatch } from "redux";

import { composeWithDevTools } from "redux-devtools-extension";

import { reducer as configWindowReducer } from "./configWindow";
import { operations as configWindowOperations } from "./configWindow";

import { reducer as cursolReducer } from "./cursol";

import { reducer as estimatorReducer } from "./estimator";
import { middleware as estimatorMiddleware } from "./estimator";
import { operations as estimatorOperations } from "./estimator";

import { reducer as timerReducer } from "./timer";
import { middleware as timerMiddleware } from "./timer";

export interface RootState {
  configWindow: {
    isActive: boolean;
    scanSpeed: number;
  };
  cursol: {
    activeColumn: number;
  };
  timer: {
    id: number | null;
    isActive: boolean;
    scanSpeed: number;
  };
  estimator: {
    id: number | null;
    result: string;
  };
}

const middlewares = [
  timerMiddleware,
  estimatorMiddleware,
];

const enhancer = composeWithDevTools(
  applyMiddleware(...middlewares),
);

export const reducer = combineReducers({
  configWindow: configWindowReducer,
  cursol: cursolReducer,
  estimator: estimatorReducer,
  timer: timerReducer,
});

export const store = createStore(
  reducer,
  enhancer,
);

export const operations = (dispatch: Dispatch) => {
  return {
    configureWindowOpen: configWindowOperations.windowOpen(dispatch),
    resetEstimator: estimatorOperations.reset(dispatch),
    sendKey: estimatorOperations.sendKey(dispatch),
    startFetchEstimatorId: estimatorOperations.fetchId(dispatch),
  };
};
