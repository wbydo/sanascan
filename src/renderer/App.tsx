import * as React from "react";
import {connect} from "react-redux";
import {Dispatch} from "redux";

import CharacterBoard from "./CharacterBoard";
import {Action, asyncIncrement} from "./actions";

interface Props {
  asyncIncrement: () => void;
}

type MapDispatchToProps = (dispatch: Dispatch<Action>) => Props;

class App extends React.Component<Props, {}> {
  public render() {
    return(
      <div>
        <CharacterBoard />
        <button type="button" onClick={this.handleOnClick}>
          async
        </button>
      </div>
    );
  }

  public handleOnClick = (event: React.MouseEvent): void => {
    return this.props.asyncIncrement();
  }
}

const mdp: MapDispatchToProps = (dispatch) => {
  return {
    asyncIncrement: () => dispatch(asyncIncrement()),
  };
};

export default connect(null, mdp)(App);
