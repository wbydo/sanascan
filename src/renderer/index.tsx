import * as React from "react";
import * as ReactDOM from "react-dom";
import { createStore, applyMiddleware } from "redux";
import { Provider } from "react-redux";
import createSagaMiddleware from "redux-saga";

import App from "./App";
import reducer from "./reducers";
import rootSaga from "./sagas";

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  reducer,
  applyMiddleware(sagaMiddleware),
);

sagaMiddleware.run(rootSaga);

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById("root"),
);
