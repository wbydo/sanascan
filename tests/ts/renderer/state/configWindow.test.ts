import { reducer } from "sanascan/renderer/state";

import { initialState } from "./initialState";

describe("isActiveが設定できる", () => {
  for (const i of [true, false]) {
    test(i.toString() + "の場合", () => {
      const isActive = i;
      const result = reducer(
        initialState,
        {
          payload: { isActive },
          type: "sanascan/configWindow/SET_ACTIVE",
        },
      );
      expect(result).toMatchSnapshot();
    });
  }

});
