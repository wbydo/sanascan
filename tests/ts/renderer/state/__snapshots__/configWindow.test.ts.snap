// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isActiveが設定できる falseの場合 1`] = `
Object {
  "configWindow": Object {
    "isActive": false,
    "scanSpeed": 0,
  },
  "cursol": Object {
    "activeColumn": 0,
  },
  "estimator": Object {
    "id": null,
    "result": "",
  },
  "timer": Object {
    "id": null,
    "isActive": false,
    "scanSpeed": 500,
  },
}
`;

exports[`isActiveが設定できる trueの場合 1`] = `
Object {
  "configWindow": Object {
    "isActive": true,
    "scanSpeed": 0,
  },
  "cursol": Object {
    "activeColumn": 0,
  },
  "estimator": Object {
    "id": null,
    "result": "",
  },
  "timer": Object {
    "id": null,
    "isActive": false,
    "scanSpeed": 500,
  },
}
`;
